unstable test TestHypoIndexDDL

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
--- FAIL: TestHypoIndexDDL (1.12s)
    result.go:49:
                Error Trace:    testkit/result.go:49
                                                        planner/core/plan_test.go:970
                Error:          Not equal:
                                expected: "[t CREATE TABLE `t` (\n  `a` int(11) DEFAULT NULL,\n  `b` int(11) DEFAULT NULL,\n  `c` int(11) DEFAULT NULL,\n  `d` int(11) DEFAULT NULL,\n  KEY `a` (`a`),\n  KEY `hypo_a` (`a`) /* HYPO INDEX */,\n  KEY `hypo_bc` (`b`,`c`) /* HYPO INDEX */\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin]\n"
                                actual  : "[t CREATE TABLE `t` (\n  `a` int(11) DEFAULT NULL,\n  `b` int(11) DEFAULT NULL,\n  `c` int(11) DEFAULT NULL,\n  `d` int(11) DEFAULT NULL,\n  KEY `a` (`a`),\n  KEY `hypo_bc` (`b`,`c`) /* HYPO INDEX */,\n  KEY `hypo_a` (`a`) /* HYPO INDEX */\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin]\n"

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -6,4 +6,4 @@
                                   KEY `a` (`a`),
                                -  KEY `hypo_a` (`a`) /* HYPO INDEX */,
                                -  KEY `hypo_bc` (`b`,`c`) /* HYPO INDEX */
                                +  KEY `hypo_bc` (`b`,`c`) /* HYPO INDEX */,
                                +  KEY `hypo_a` (`a`) /* HYPO INDEX */
                                 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin]
                Test:           TestHypoIndexDDL
                Messages:       sql:show create table t, args:[]
```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/executor/show.go
+++ b/executor/show.go
@@ -1120,9 +1120,14 @@ func constructResultOfShowCreateTable(ctx sessionctx.Context, dbName *model.CISt
 		schemaName := dbName.L
 		tblName := tableInfo.Name.L
 		if hypoIndexes[schemaName] != nil && hypoIndexes[schemaName][tblName] != nil {
+			hypoIndexList := make([]*model.IndexInfo, 0, len(hypoIndexes[schemaName][tblName]))
 			for _, index := range hypoIndexes[schemaName][tblName] {
-				publicIndices = append(publicIndices, index)
+				hypoIndexList = append(hypoIndexList, index)
 			}
+			sort.Slice(hypoIndexList, func(i, j int) bool { // to make the result stable
+				return hypoIndexList[i].Name.O < hypoIndexList[j].Name.O
+			})
+			publicIndices = append(publicIndices, hypoIndexList...)
 		}
 	}
 	if len(publicIndices) > 0 {

